src/main/java/com/capitalone/dashboard/collector/CollectorTaskWithGenericItem.java

```diff
@@ -64,7 +64,7 @@ public CollectorTaskWithGenericItem(TaskScheduler taskScheduler, String collecto
             gci.setProcessTime(System.currentTimeMillis());
             genericCollectorItemRepository.save(gci);
         });
-        return collectorItemBuildIds;
+        return Collections.unmodifiableMap(collectorItemBuildIds);
```

Comment:
I like the idea of immutable map however this might break some downstream functionalities. Hence please revert to original

Reply:
I understand :), reverting now.

One thing you may want to consider is that this change was made from the bug pattern  [MixedMutabilityReturnType](https://errorprone.info/bugpattern/MixedMutabilityReturnType) in mind because it can also return a static, immutable empty list. Probably in this case you would not need to modify the list if it is empty, but just something to consider.